tests/unit/test_row_noise.py

```diff
@@ -45,12 +45,47 @@ def test_omission(dummy_data):
     assert set(noised_data1.columns) == set(dummy_data.columns)
     assert (noised_data1.dtypes == dummy_data.dtypes).all()
 
+
+def test_do_not_respond(mocker, dummy_data):
+    config = get_configuration()[DATASETS.census.name][Keys.ROW_NOISE][
+        NOISE_TYPES.do_not_respond.name
+    ]
+    mocker.patch(
+        "pseudopeople.noise_functions._get_census_do_not_respond_demographic_probabilities",
+        side_effect=(lambda *_: config[Keys.ROW_PROBABILITY]),
+    )
+    dataset_name_1 = DATASETS.census.name
+    dataset_name_2 = DATASETS.acs.name
+    noised_data1 = NOISE_TYPES.do_not_respond(dataset_name_1, dummy_data, config, RANDOMNESS)
```

Comment:
This doesn't fully test the functionality you've implemented. Let's talk tomorrow morning about how we can manage that. I have some ideas involving using different dummy datasets containing only (for example) black males less than 5 years old, or Asian females who are 30-49.